### ISSUE TITLE
Interruption Handling Notifications not working

### ISSUE THREAD

**Issue by Kedar-27** [analogcode/Swift-Radio-Pro#128]
`    
private func setupNotifications()
        { `

   `     let notificationCenter = NotificationCenter.default
        notificationCenter.addObserver(self, selector: #selector(handleInterruption), name: AVAudioSession.interruptionNotification, object: nil)
        notificationCenter.addObserver(self, selector: #selector(handleRouteChange), name: AVAudioSession.routeChangeNotification, object: nil)
        notificationCenter.addObserver(self, selector: #selector(playerDidFinishPlaying), name: .AVPlayerItemDidPlayToEndTime, object: nil)

        notificationCenter.addObserver(self, selector: #selector(playerDidRecieveError(_:)), name: .AVPlayerItemNewErrorLogEntry, object: nil)

         notificationCenter.addObserver(self, selector: #selector(playerDidRecieveError(_:)), name: .AVPlayerItemFailedToPlayToEndTime , object: nil)

        notificationCenter.addObserver(self, selector: #selector(playerDidRecieveError(_:)), name: .AVPlayerItemPlaybackStalled , object: nil)`
`}`

I added 3 more notifications for interruption handling, but seems not to be working please help with this issue @analogcode

**Comment by fethica** [comment]
@Kedar-27 Can you add more context on why you'd like to add the notifications? The interruptions are handled by default, and you can use the delegate interface of the RadioPlayer to add custom behaviour.

**Comment by Kedar-27** [comment]
> @Kedar-27 Can you add more context on why you'd like to add the notifications? The interruptions are handled by default, and you can use the delegate interface of the RadioPlayer to add custom behavior.

Yes, we have delegates, but they all detect the error at the start of playback. I wanted more precise error handling during playback
